src/telemetry/CMakeLists.txt

```diff
@@ -26,3 +26,10 @@ if (TELEMETRY_INSTALL_TARGETS)
 	install(TARGETS telemetry LIBRARY DESTINATION ${INSTALL_DIR_LIB})
 	install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/ DESTINATION ${INSTALL_DIR_INCLUDE})
 endif()
+
+if (TELEMETRY_ENABLE_TESTS)
+	add_executable(testTelemetry ${TELEMETRY_SOURCE_FILES})
+	target_compile_definitions(testTelemetry PRIVATE TELEMETRY_ENABLE_TESTS)
+	target_link_libraries(testTelemetry telemetry GTest::gtest_main)
```

Comment:
Is it really necessary to link it against `telemetry` library as we build independent binary that contains all functions?